op-devstack/sysgo/superroot.go

```diff
@@ -0,0 +1,445 @@
+			o.P().Require().NoError(err, "invalid OPCM ABI")
+			chainOps := devkeys.ChainOperatorKeys(l1ChainID.ToBig())
+			proposer, err := o.keys.Address(chainOps(devkeys.ProposerRole))
+			o.P().Require().NoError(err, "must have configured proposer")
+			challenger, err := o.keys.Address(chainOps(devkeys.ChallengerRole))
+			o.P().Require().NoError(err, "must have configured challenger")
+
+			opcmAddr := o.wb.output.ImplementationsDeployment.OpcmImpl
+			contract := batching.NewBoundContract(&opcmABI, opcmAddr)
+			migrateInput := bindings.OPContractsManagerInteropMigratorMigrateInput{
+				UsePermissionlessGame: true,
+				StartingAnchorRoot: bindings.Proposal{
+					Root:             common.Hash(superRoot),
+					L2SequenceNumber: big.NewInt(int64(header.Time)),
+				},
+				GameParameters: bindings.OPContractsManagerInteropMigratorGameParameters{
+					Proposer:         proposer,
+					Challenger:       challenger,
+					MaxGameDepth:     big.NewInt(73),
+					SplitDepth:       big.NewInt(30),
+					InitBond:         big.NewInt(0),
+					ClockExtension:   10800,
+					MaxClockDuration: 302400,
+				},
+				OpChainConfigs: opChainConfigs,
+			}
+			migrateCall := contract.Call("migrate", migrateInput)
+			migrateCallData, err := migrateCall.Pack()
+			require.NoError(err)
+
+			l1PAOKey, err := o.keys.Secret(chainOps(devkeys.L1ProxyAdminOwnerRole))
+			require.NoError(err, "must have configured L1 proxy admin owner private key")
+			transactOpts, err := bind.NewKeyedTransactorWithChainID(l1PAOKey, l1ChainID.ToBig())
+			require.NoError(err, "must have transact opts")
+			transactOpts.Context = t.Ctx()
+
+			t.Log("Deploying delegate call proxy contract")
+			// The DelegateCallProxy is used to simulate a GnosisSafe proxy that satisfies the delegatecall requirement of the OPCM.
+			delegateCallProxy, proxyContract := deployDelegateCallProxy(t, transactOpts, client, l1pao)
+			oldSuperchainProxyAdminOwner := getOwner(t, w3Client, superchainProxyAdmin)
+			transferOwnership(t, l1PAOKey, client, superchainProxyAdmin, delegateCallProxy)
+
+			oldDisputeGameFactories := make(map[eth.ChainID]common.Address)
+			for i, opChainConfig := range opChainConfigs {
+				chainOpsForL2 := devkeys.ChainOperatorKeys(l2ChainIDs[i].ToBig())
+				l1PAOKeyForL2, err := o.keys.Secret(chainOpsForL2(devkeys.L1ProxyAdminOwnerRole))
+				require.NoError(err, "must have configured L1 proxy admin owner private key")
+
+				var portal common.Address
+				require.NoError(
+					w3Client.Call(
+						w3eth.CallFunc(opChainConfig.SystemConfigProxy, optimismPortalFn).Returns(&portal),
+					))
+				portalProxyAdmin := getProxyAdmin(t, w3Client, portal)
+				transferOwnership(t, l1PAOKeyForL2, client, portalProxyAdmin, delegateCallProxy)
+
+				dgf := getDisputeGameFactory(t, w3Client, portal)
+				transferOwnership(t, l1PAOKeyForL2, client, dgf, delegateCallProxy)
+				oldDisputeGameFactories[l2ChainIDs[i]] = dgf
+			}
+
```

Comment:
All these ownership transfers feels like hard work. As a follow up I wonder if either:
1. we just have a 1-of-1 gnosis safe deployed and used as the proxy owner as part of deploying a new chain at genesis - seems crazy to deploy a system in a configuration that isn't compatible with OPCM
2. we could use EIP7702 to allow the EOA to do a delegate call

I feel like either option is more of a follow up project than a follow up PR though. This complexity is encapsulated here well so I don't think it's a huge priority to remove.

Reply:
My understanding is that to do `1` it'll also require some cooperation with kurtosis. So indeed, it's more of a project-level effort to do. This is being tracked by https://github.com/ethereum-optimism/optimism/issues/15882. We'll be able to remove the transfers here once that's fixed.
`2` is a good bad idea. Long term we would still want a proper 1 of 1 safe setup so we can rehearse OPCM operations locally using kurtosis.